aws/config_service_test.go

```diff
@@ -0,0 +1,279 @@
+	"github.com/gruntwork-io/cloud-nuke/util"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestListConfigServiceRules(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+
+	configRuleName := createConfigServiceRule(t, region)
+	defer deleteConfigServiceRule(t, region, configRuleName, false)
+
+	configServiceRuleNames, err := getAllConfigRules(session, time.Now(), config.Config{})
+	require.NoError(t, err)
+	assert.Contains(t, configServiceRuleNames, configRuleName)
+}
+
+func TestNukeConfigServiceRuleOne(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+
+	configRuleName := createConfigServiceRule(t, region)
+	defer deleteConfigServiceRule(t, region, configRuleName, false)
+
+	configServiceRuleNames := []string{configRuleName}
+
+	require.NoError(
+		t,
+		nukeAllConfigServiceRules(session, configServiceRuleNames),
+	)
+
+	assertConfigServiceRulesDeleted(t, region, configServiceRuleNames)
+}
+
+func TestNukeConfigServiceRuleMoreThanOne(t *testing.T) {
+	t.Parallel()
+
+	region, err := getRandomRegion()
+	require.NoError(t, err)
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+
+	configServiceRuleNames := []string{}
+	for i := 0; i < 3; i++ {
+		configServiceRuleName := createConfigServiceRule(t, region)
+		defer deleteConfigServiceRule(t, region, configServiceRuleName, false)
+		configServiceRuleNames = append(configServiceRuleNames, configServiceRuleName)
+	}
+
+	require.NoError(
+		t,
+		nukeAllConfigServiceRules(session, configServiceRuleNames),
+	)
+
+	assertConfigServiceRulesDeleted(t, region, configServiceRuleNames)
+}
+
+// Test helpers
+
+// ensureConfigurationRecorderExistsInRegion is a convenience method to be used during testing
+// since you cannot create a custom configuration rule unless you already have a configuration
+// recorder in the target region
+func ensureConfigurationRecorderExistsInRegion(t *testing.T, region string) (string, error) {
+	session, err := session.NewSession(&aws.Config{Region: aws.String(region)})
+	require.NoError(t, err)
+
+	configService := configservice.New(session)
+
+	param := &configservice.DescribeConfigurationRecordersInput{}
+
+	output, lookupErr := configService.DescribeConfigurationRecorders(param)
+	require.NoError(t, lookupErr)
+
+	// If we have no recorders in the region, create one - otherwise we have no more work to do
+	if len(output.ConfigurationRecorders) == 0 {
+		configRecorderName := createConfigurationRecorder(t, region)
+		require.NotEqual(t, "", configRecorderName)
+		return configRecorderName, nil
+	}
+
+	return aws.StringValue(output.ConfigurationRecorders[0].Name), nil
```

Comment:
Noticed that `ensureConfigurationRecorderExistsInRegion()` in all cases return `nil` as error

Reply:
Good catch - fixed in [a488a7d](https://github.com/gruntwork-io/cloud-nuke/pull/405/commits/a488a7dba2e73ec2706fb3bf60b4dc4f9b096efa)